backend/db.js

```diff
@@ -0,0 +1,106 @@
+import { DataSource } from 'apollo-datasource'
+import crypto from 'crypto'
+
+export class Post {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString('hex')
+        Object.assign(this, data)
+    }
+}
+
+export class User {
+    constructor(data) {
+        this.id = crypto.randomBytes(16).toString('hex')
+        Object.assign(this, data)
+    }
+}
+
+export class InMemoryDataSource extends DataSource {
+    constructor() {
+        super()
+        this.posts = []
+        this.users = []
+        this.initDummyUsers()
+        this.initDummyPosts()
+    }
+
+    initialize(...args) {
+        //Warum wird diese Methode so oft ausgeführt?
+    }
+
+    createPost(data) {
+        const author = this.users.find(user => user.name === data.post.author.name)
+        const newPost = new Post({
+            title: data.post.title,
+            votes: [],
+            author: author
+        })
+        this.posts.push(newPost)
+        const newUsers = this.users.map(user => user.id === newPost.author.id ? this.addPostToAuthor(newPost, user) : user)
```

Comment:
I would suggest to use only one foreign-key in order to avoid redundancy. Another group already had a bug in which they setup bi-directional foreign keys in a wrong way.

You only need only one foreign key, so `post.author` or `user.posts`.